### DISCUSSION TITLE
Parent and child blocks

### DISCUSSION THREAD

**Discussion by cata008** [Azure/ipam#196] [Ideas]
Hi @DCMattyG,

Would it be possible to have parent and child blocks in IPAM? I remember in Infoblox you'd have the concept of networks and network containers. Once a network would contain other networks it would instantly become a network container.

The reason why this came to mind is because I have one space containing all environments (dev,test etc). Within this space, I have created a bunch of /16 as per the existing networking infrastructure splitting. They are all part of 10/8. However, it would be nice to have all these chunks as child blocks and whenever I need the next /16 to just carve it out from the parent block. Would this be something feasible?

Thanks,
Catalin

**Comment by DCMattyG** [comment]
Hi @cata008, while anything is possible, this will require a lot of rework as the current design all revolves around a single depth of space/block. I can definitely look into the level of effort here as I do agree this would be nice, but know that this ask will take some time for sure.

Always happy to explore the possibilities of what can be done for you 😉

**Reply by cata008** [reply]
@DCMattyG thanks for looking into it. Maybe this can be v2 at some point :)
